s3transfer/futures.py

```diff
@@ -128,8 +128,9 @@ def set_result(self, result):
         Implies that the TransferFuture succeeded.
         """
         with self._lock:
-            self._result = result
-            self._status = 'success'
+            if self._exception is None:
```

Comment:
If one thread came along and set an exception. Another thread that may not have been signaled that there was a failure may come along and clobber the exception with a success. We need to keep whatever the exception that was thrown.